### ISSUE TITLE
Error with dict literal as TypedDict argument with metaclass

### ISSUE THREAD

**Issue by olavmo-sikt** [microsoft/pyright#9067]
**Describe the bug**

When instantiating a class which has a metaclass overriding `__call__`, Pyright fails to see that a dict literal is compatible with a TypedDict argument.

**Code or Screenshots**

Code sample in [pyright playground](https://pyright-play.net/?code=JYWwDg9gTgLgBDAnmYA7A5gKEwYwDYCGAzkXAAoFQEhEAUSKGAdACrICmAJgCLA4wBKAFyY4YuADNg7PJyFw0MbPmKkAYhAgBZdjAL0Ow0eM7sJcAPoWcBPHiu18RADRwAVJXQv3bgNYB3TyIjcVC4IgBXMHYoWgEmKxs7Bw8oL1c3P0C04OVCEjgNCFoQXQIVEgBeIp09EJMzSws0YBgHIhkJVzBKaiJ5CioaerC4HpJsItoAbwAiKRlOWfkAFgAmAF8BTCA):

```python
import typing

class Params(typing.TypedDict):
    field: int

class FooMeta(type):
    def __call__(cls, *args, **kwargs):
        super().__call__(*args, **kwargs)

class Foo(metaclass=FooMeta):
    def __init__(self, params: Params):
        pass

Foo({"field": 42})
```

The above call to `Foo(...)` fail type checking with the following error:

```
Argument of type "dict[str, int]" cannot be assigned to parameter "params" of type "Params" in function "__init__" \
  "dict[str, int]" is not assignable to "Params"  (reportArgumentType)
```

Removing `__call__` from `FooMeta` allows Pyright to see that the dict literal is compatible.

**VS Code extension or command-line**

Version in VS code:
> Pylance language server 2024.9.2 (pyright version 1.1.378, commit ce64adc1)

Also fails with latest Pyright (version 1.1.381) in Pyright playground.
